src/stores/ShardDetail.ts

```diff
@@ -85,74 +237,114 @@ export const getInitialState = (
                 'Shard List Set'
             );
         },
-        getShardStatusColor: (catalogNamespace) => {
-            const { shards } = get();
+        getTaskShards: (catalogNamespace, shards) => {
+            return shards.length > 0
+                ? shards.filter(({ spec }) => {
+                      const labels = spec.labels ? spec.labels.labels : [];
 
-            if (shards.length > 0) {
-                const selectedShard = shards.find(({ spec }) =>
-                    spec.id ? spec.id.includes(catalogNamespace) : undefined
+                      const taskName = labels?.find(
+                          (label) => label.name === 'estuary.dev/task-name'
+                      )?.value;
+
+                      return taskName === catalogNamespace;
+                  })
+                : [];
+        },
+        getTaskShardDetails: (taskShards) => {
```

Comment:
`getTaskShardDetails` returns the information required to perform a calculation of aggregate shard status as well as relay the statuses of all shards associated with the specific task. Its only use is within the `EntityStatus` component.